### DISCUSSION TITLE
Disabling of default providers for certain SPIs

### DISCUSSION THREAD

**Discussion by kareemserry** [keycloak/keycloak#10223] [Archive]
Hello all, 

I am currently trying to port over some custom modules and their respective configuration over to the quarkus distribution. I am however facing some challenges with mapping the existing configuration to the new format.

for example, this bit of configration code in the old standalone.xml files allowed me to disable the default freemarker provider and have it replaced by my custom provider. 

``` xml
<spi name="emailTemplate">
   <provider name="freemarker" enabled="false"/>
   <provider name="custom-freemarker" enabled="true"/>
</spi>
```
            
Based on https://github.com/keycloak/keycloak-documentation/pull/1408 I would expect one of these to work on the new distribution, none however do.

``` properties
spi-emailTemplate-freemarker-enabled=false
spi-emailTemplate-custom-freemarker-enabled=true
```
``` properties
spi-emailTemplate-provider=custom-freemarker
```

**Comment by pedroigor** [comment]
@kareemserry Could you please check if this new guide helps you ? See https://github.com/keycloak/keycloak/pull/10401.
